File: 2784-check-if-array-is-good/2784-check-if-array-is-good.cs
public class Solution {
    public bool IsGood(int[] nums) {
        if (nums.Length == 1)
        {
            return false;
        }

        int n = nums.Length - 1;

        int[] Freq = new int[n];
        foreach (int i in nums)
        {
            if (i > n)
            {
                return false;
            }

            Freq[i - 1]++;
        }

        bool isAppearOnce = true;
        for (int i = 0; i < Freq.Length - 1; i++)
        {
            if (Freq[i] != 1)
            {
                isAppearOnce = false;
            }
        }

        bool isNAppearTwice = Freq[n - 1] == 2;
        return isAppearOnce && isNAppearTwice;
    }
}

File: 2784-check-if-array-is-good/2784-check-if-array-is-good.java
class Solution {
    public boolean isGood(int[] nums) {
        if (nums.length == 1) {
            return false;
        }

        int n = nums.length - 1;

        int[] freq = new int[n]; // use 1 indexing so that freq[0] is the freq when nums = 1
        for (int i : nums) {
            if (i > n) {
                return false; // becomes invalid
            }

            freq[i - 1]++;
        }

        boolean isAppearOnce = true;
        for (int i = 0; i < freq.length - 1; i++) {
            if (freq[i] != 1) {
                isAppearOnce = false;
            }
        }

        boolean isNAppearTwice = freq[n - 1] == 2;
        return isAppearOnce && isNAppearTwice;
    }
}